bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/LaunchBarListener.java

```diff
@@ -98,25 +100,29 @@ private void update(String newTarget)
 					{
 						// get current target
 						String currentTarget = new SDKConfigJsonReader((IProject) project).getValue("IDF_TARGET"); //$NON-NLS-1$
-						final String jtag_device_id = activeConfig.getAttribute("org.eclipse.cdt.debug.gdbjtag.core.jtagDeviceId", ""); //$NON-NLS-1$ //$NON-NLS-2$
-						if ((activeConfig.getAttribute(IDFLaunchConstants.FLASH_OVER_JTAG, false) && !jtagIgnored)
+						final String jtag_device_id = activeConfig
+								.getAttribute("org.eclipse.cdt.debug.gdbjtag.core.jtagDeviceId", ""); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
Wouldn't it be better to move the constant strings to separate properties or interface?

Reply:
fixed, by providing better way to check if selected configuration is of debug type